### PR TITLE
Working CBF with bdk master branch 

### PR DESCRIPTION
**Pull request by andreasgriffin**
see https://github.com/bitcoindevkit/bdk-ffi/pull/207#issuecomment-1507486619

sorry for the confusion with the closed PR.

### CODE REVIEW

File: bdk-ffi/src/blockchain.rs [hunk 1]
```diff
@@ -225,6 +231,6 @@ pub enum BlockchainConfig {
     Esplora { config: EsploraConfig },
     /// Bitcoin Core RPC client
     Rpc { config: RpcConfig },
-    /// Nakamoto client
-    CBF { config: CBFConfig },
+    /// CompactFilters
+    Cbf { config: CompactFiltersConfig },
```
Comment (thunderbiscuit) [inline bdk-ffi/src/blockchain.rs]: Let's call it `CompactFilters` so as to keep in line with the naming of the other fields (which reflect the enum variant in the `AnyBlockchainConfig` type).

```rust
/// CompactFilters
CompactFilters { config: CompactFiltersConfig },
```

File: bdk-ffi/src/blockchain.rs [hunk 2]
```diff
@@ -43,34 +43,36 @@ impl Blockchain {
...
+            BlockchainConfig::Cbf { config } => {
+
+                let mut peers = Vec::new();
+                for address in config.addresses {
+                    peers.push(BitcoinPeerConfig{
+                        address:address,
+                        socks5:None,
+                        socks5_credentials:None
+                    });
+                }
```
Comment (thunderbiscuit) [inline bdk-ffi/src/blockchain.rs]: As we prepare this PR for final merge we should consider expanding the full `peers: Vec<BitcoinPeerConfig>` type.

Comment (thunderbiscuit) [reply bdk-ffi/src/blockchain.rs]: For now I think this is good. I just want to keep track of it.

File: bdk-ffi/src/blockchain.rs [hunk 3]
```diff
@@ -217,6 +215,14 @@ pub struct RpcConfig {
     pub sync_params: Option<RpcSyncParams>,
 }
 
+
+pub struct CompactFiltersConfig {
+    pub addresses: Vec<String>,
+    pub network: Network,
+    pub storage_dir: String,
+    pub skip_blocks: u32,
+}
```
Comment (thunderbiscuit) [inline bdk-ffi/src/blockchain.rs]: Let's add the inline docs from the [Rust docs](https://docs.rs/bdk/0.27.1/bdk/blockchain/compact_filters/struct.CompactFiltersBlockchainConfig.html) on the `CompactFilterConfig` type.

File: bdk-ffi/src/bdk.udl [hunk 4]
```diff
@@ -161,12 +157,23 @@ dictionary RpcConfig {
...
+};
+
+
 [Enum]
 interface BlockchainConfig {
   Electrum(ElectrumConfig config);
   Esplora(EsploraConfig config);
   Rpc(RpcConfig config);
-  CBF(CBFConfig config);
+  Cbf(CompactFiltersConfig config);
```
Comment (thunderbiscuit) [inline bdk-ffi/src/bdk.udl]: See comment below about renaming.

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
Looks good! Do you mind running `cargo fmt` so as to clean up the few syntax issues?

Also, I'm trying to test locally on testnet but I don't have a peer address I can use in the `address` field of the config. Do you mind sharing one if you have any, or let me know how you tested this? I think the previous implementation went to some default peer if none was provided, but here we'll need at least one.

**Comment by andreasgriffin** [comment]
> Also, I'm trying to test locally on testnet but I don't have a peer address I can use in the `address` field of the config. Do you mind sharing one if you have any, or let me know how you tested this? I think the previous implementation went to some default peer if none was provided, but here we'll need at least one.

Hi,  I tested it all with a local regtest/testnet/mainnet node by enabling (bitcoin.conf)
```
peerblockfilters=1
blockfilterindex=1
```
Regtest is easiest, because it is fast.  Testnet needs a lot of time (way more than mainnet) to sync (and sometimes crashes in the middle). The bdk CBF implementation is clearly still "experimental".


I've done the requested changes.

**Comment by andreasgriffin** [comment]
It seems the blockchain.broadcast method (for cbf) is defunct. 
image
The node does not receive the tx.  When I broadcast it via rpc or the [bitcoin p2p](https://github.com/andreasgriffin/bitcoin-p2p) messages, the tx is accepted in the nodes mempool.

**Comment by andreasgriffin** [comment]
Hi  @thunderbiscuit :  Can you please update your master to the bitcoindevkit/master, so the changed files in the PR is useful.

**Comment by thunderbiscuit** [comment]
I pulled the latest changes on `bdk-ffi/master` into my master and rebased this branch on that. It was a... total mess. I expect it will break this PR hard. I'm not sure if it's worth keeping up as a PR until we are really ready to put work into this (once bdk 1.0 is release with the compact block filters capability). Feel free to keep this PR open though! I just don't want to promise things I won't have resources for.

**Comment by thunderbiscuit** [comment]
@andreasgriffin how do you feel about starting fresh and opening a PR on bdk-ffi directly from your fork and cbf branch? This would allow you to have better control over the commit history, and that way you can keep it up to date with `master` on bdk-ffi if you have the resources/need to do that.

Let me know if that's something you'd like to take on; once you open the PR I can close bitcoindevkit/bdk-ffi#207 in favour of yours.

**Comment by andreasgriffin** [comment]
@thunderbiscuit : Yes opening a PR on bdk-ffi directly  (even though it won't get merged before bdk 1.0), is better. I'll do that.

> (once bdk 1.0 is release with the compact block filters capability)

Do you have any idea, when a CBF directory will be added to https://github.com/bitcoindevkit/bdk/tree/master/crates

**Comment by thunderbiscuit** [comment]
No unfortunately I don't have a good time estimate. I know it's on the roadmap, but the 1.0 has taken a lot more work than initially anticipated so I don't really know where they stand with the CBF just yet.

Feel free to ask on the #development channel of the Discord server! Are you with us on the server yet? I hope so! Let me know if you need a fresh invite.

**Comment by andreasgriffin** [comment]
Thanks, I will ask on the #development channel.

**Pull request closed without merging**